src/context.rs

```diff
@@ -566,6 +569,26 @@ impl PySessionContext {
         Ok(())
     }
 
+    /// Construct datafusion dataframe from Arrow Table
+    pub fn register_table_provider(
+        &mut self,
+        name: &str,
+        provider: Bound<'_, PyAny>,
+        py: Python,
+    ) -> PyResult<()> {
+        if provider.hasattr("__datafusion_table_provider__")? {
```

Comment:
Should we raise when it doesn't have this attribute, it might give the wrong impression for users that they registered it but nothing happened

Reply:
Good point! Added.